src/suites/cts/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,168 @@
+export const description = `
+copy imageBitmap To texture tests.
+`;
+
+import { TestGroup } from '../../framework/index.js';
+
+import { GPUTest } from './gpu_test.js';
+
+class F extends GPUTest {
+  calculateRowPitch(width: number, bytesPerPixel: number): number {
+    const bytesPerRow = width * bytesPerPixel;
+    // Rounds up to a multiple of 256 according to WebGPU requests.
+    return (((bytesPerRow - 1) >> 8) + 1) << 8;
+  }
+
+  checkCopyImageBitmapResult(
+    src: GPUBuffer,
+    expected: ArrayBufferView,
+    width: number,
+    height: number,
+    bytesPerPixel: number
+  ): void {
+    const exp = new Uint8Array(expected.buffer, expected.byteOffset, expected.byteLength);
+    const rowPitch = this.calculateRowPitch(width, bytesPerPixel);
+    const dst = this.createAndMapBufferForReadingContents(src, rowPitch * height);
+
+    this.eventualAsyncExpectation(async niceStack => {
+      const actual = new Uint8Array(await dst.mapReadAsync());
+      const check = this.checkBufferWithRowPitch(
+        actual,
+        exp,
+        width,
+        height,
+        rowPitch,
+        bytesPerPixel
+      );
+      if (check !== undefined) {
+        niceStack.message = check;
+        this.rec.fail(niceStack);
+      }
+      dst.destroy();
+    });
+  }
+
+  checkBufferWithRowPitch(
+    actual: Uint8Array,
+    exp: Uint8Array,
+    width: number,
+    height: number,
+    rowPitch: number,
+    bytesPerPixel: number
+  ): string | undefined {
+    const size = exp.byteLength;
+    const lines = [];
+    let failedPixels = 0;
+    for (let i = 0; i < height; ++i) {
+      const bytesPerRow = width * bytesPerPixel;
+      for (let j = 0; j < bytesPerRow; ++j) {
+        const indexExp = j + i * bytesPerRow;
+        const indexActual = j + rowPitch * i;
+        if (actual[indexActual] !== exp[indexExp]) {
+          if (failedPixels > 4) {
+            break;
+          }
+          failedPixels++;
+          lines.push(`at [${indexExp}], expected ${exp[indexExp]}, got ${actual[indexActual]}`);
+        }
+      }
+      if (failedPixels > 4) {
+        lines.push('... and more');
+        break;
+      }
+    }
+    return this.handleErrorMessage(actual, exp, size, failedPixels, lines);
+  }
+
+  // Using drawImage to extract imageBitmap content.
+  imageBitmapToData(imageBitmap: ImageBitmap): ArrayBufferView | undefined {
+    const imageCanvas = document.createElement('canvas');
+    imageCanvas.width = imageBitmap.width;
+    imageCanvas.height = imageBitmap.height;
+
+    const imageCanvasContext = imageCanvas.getContext('2d');
+    this.expect(imageCanvasContext != null);
+    if (!imageCanvasContext) {
+      return undefined;
```

Comment:
`throw new Error(...)` instead of `return undefined` so that the return type of the function doesn't have to have `| undefined` in it.

Reply:
Oh, and the 'expect' won't be needed either. An exception will cause the test to fail.

Reply:
Done.